js/src/admin.js

```diff
@@ -2,6 +2,67 @@
  * @package Polylang
  */
 
+window.pll = window.pll || {};
+
+/**
+ * Calls the passed callable when the DOM is ready.
+ *
+ * @param {callable} fn - Function to call when the DOM is ready.
+ */
+window.pll.dom_ready = ( fn ) => {
+	if ( document.readyState !== 'loading' ) {
+		fn();
+	} else {
+		document.addEventListener( 'DOMContentLoaded', fn );
+	}
+};
+
+/**
+ * Settings: manages AJAX calls attached to `.pll-ajax-button` buttons.
+ */
+window.pll.settings_ajax_button = () => {
+	document.querySelectorAll( '.pll-ajax-button' ).forEach( ( el ) => {
+		el.addEventListener( 'click', ( event ) => {
+			const button    = event.target;
+			const action    = button.getAttribute( 'data-action' );
+			const nonce     = button.getAttribute( 'data-nonce' );
+			const row       = button.closest( 'tr' );
+			const errorElms = row.querySelectorAll( '.pll-error-message-text' );
+
+			if ( ! action || ! nonce || ! row || ! errorElms.length ) {
+				return;
+			}
+
+			const urlParams = { 'action': action, '_pll_nonce': nonce, 'pll_ajax_settings': 1 };
+			row.querySelectorAll( '[data-name]' ).forEach( ( el ) => {
+				urlParams[ el.getAttribute( 'data-name' ) ] = el.value;
+			} );
+			const url = wp.url.addQueryArgs( ajaxurl, urlParams );
+
+			button.setAttribute( 'disabled', 'disabled' );
+			row.classList.remove( 'notice-success', 'notice-error', 'notice-alt' );
+
+			fetch( url ).then( ( response ) => {
+				button.removeAttribute( 'disabled' );
+				return response.json();
+			} ).then( ( json ) => {
+				if ( json.success ) {
+					row.classList.add( 'notice-success', 'notice-alt' );
```

Comment:
This kind of logic could benefit from a `toggle` function along `button.removeAttribute( 'disabled' );`.

Reply:
I didn't went through, it adds code instead of removing some.

Reply:
🤔 we can use `toggleAttribute( 'disabled')` everywhere instead of `setAttribute` or `removeAttribute`
See https://developer.mozilla.org/en-US/docs/Web/API/Element/toggleAttribute

Reply:
Nope, I'm against any `toggle*()`, it's too unpredictable: when you use it don't know if it will add or remove the attribute.